collectors/framework/framework.go

```diff
@@ -256,6 +256,23 @@ func (a *AvroDatum) transform(nodeInfo collectors.NodeInfo) (producers.MetricsMe
 		return pmm, err
 	}
 
+	// Convert message-level labels to datapoint-level tags
+	labels := pmm.Dimensions.Labels
+	if labels != nil {
+		for n, d := range pmm.Datapoints {
+			tt := make(map[string]string)
+			// TODO(philip) account for tag/label collision
```

Comment:
It seems like this TODO is a requirement.  If there is a collision we should probably log it at the least, and possibly return an error.